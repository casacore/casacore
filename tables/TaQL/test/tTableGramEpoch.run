#!/bin/sh
#-----------------------------------------------------------------------------
# Script to test the TableGram and TableParse class.  All files generated will
# be deleted on exit.
#=============================================================================

# Create a symlink to achieve that casacore_memcheck (if used) will use
# that name 
rm -f tTableGramEpoch
ln -s tTableGram tTableGramEpoch

# Execute all kind of TaQL commands involving epochs.
# Check the result using showtableinfo (leave out version line).

# Create a table with an Epoch column.
$casa_checktool ./tTableGramEpoch 'select 4mar53 giving tTableGramEpoch_tmp.tab'
../../apps/taql -pc 'select * from tTableGramEpoch_tmp.tab'
../../apps/showtableinfo in=tTableGramEpoch_tmp.tab colkey=1 | grep -v 'showtableinfo:'

# Same with unit seconds.
$casa_checktool ./tTableGramEpoch 'select 4mar53 s giving tTableGramEpoch_tmp.tab'
../../apps/taql -pc 'select * from tTableGramEpoch_tmp.tab'
../../apps/showtableinfo in=tTableGramEpoch_tmp.tab colkey=1 | grep -v 'showtableinfo:'

# Insert another row.
$casa_checktool ./tTableGramEpoch 'insert into tTableGramEpoch_tmp.tab values (04-03-1963)'
./tTableGram 'select * from tTableGramEpoch_tmp.tab'
../../apps/taql -pc 'select * from tTableGramEpoch_tmp.tab'

# Update a row.
$casa_checktool ./tTableGramEpoch 'update tTableGramEpoch_tmp.tab set Col_1=1958-05-11 where Col_1=4-mar-1963'
./tTableGram 'select * from tTableGramEpoch_tmp.tab'
../../apps/taql -pc 'select * from tTableGramEpoch_tmp.tab'

# Create using SELECT with a given data type.
$casa_checktool ./tTableGramEpoch 'select 34447 as TIME TIME giving tTableGramEpoch_tmp.tab'
../../apps/taql -pc 'select * from tTableGramEpoch_tmp.tab'
../../apps/showtableinfo in=tTableGramEpoch_tmp.tab colkey=1 | grep -v 'showtableinfo:'

# Create using CREATE TABLE
$casa_checktool ./tTableGramEpoch 'create table tTableGramEpoch_tmp.tab TIME TIME'
../../apps/showtableinfo in=tTableGramEpoch_tmp.tab colkey=1 | grep -v 'showtableinfo:'
$casa_checktool ./tTableGramEpoch 'create table tTableGramEpoch_tmp.tab TIME EPOCH [UNIT="s"]'
../../apps/showtableinfo in=tTableGramEpoch_tmp.tab colkey=1 | grep -v 'showtableinfo:'


# Remove the symlink
rm -f tTableGramEpoch
