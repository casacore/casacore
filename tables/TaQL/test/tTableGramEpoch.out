select 4mar53 giving tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  Col_1
 34440
using style python select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  Col_1
04-Mar-1953

Structure of table /Users/diepen/casa/masktql/build/dbg/tables/TaQL/test/tTableGramEpoch_tmp.tab
------------------ 
1 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=StandardStMan  bucketsize=256
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=126
  Col_1 double   scalar unit=[d] measure=epoch,UTC

Keywords of main table 
----------------------
  Column Col_1
    MEASINFO: {
      type: String "epoch"
      Ref: String "UTC"
    }
    QuantumUnits: String array with shape [1]
      [d]

select 4mar53 s giving tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  Col_1
 2.97562e+09
using style python select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  Col_1
04-Mar-1953

Structure of table /Users/diepen/casa/masktql/build/dbg/tables/TaQL/test/tTableGramEpoch_tmp.tab
------------------ 
1 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=StandardStMan  bucketsize=256
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=126
  Col_1 double   scalar unit=[s] measure=epoch,UTC

Keywords of main table 
----------------------
  Column Col_1
    MEASINFO: {
      type: String "epoch"
      Ref: String "UTC"
    }
    QuantumUnits: String array with shape [1]
      [s]

insert into tTableGramEpoch_tmp.tab values (04-03-1963)
    has been executed
    insert result of 1 rows
1 selected columns:  Col_1
 3.29115e+09
select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 2 rows
1 selected columns:  Col_1
 2.97562e+09
 3.29115e+09
using style python select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 2 rows
1 selected columns:  Col_1
04-Mar-1953
04-Mar-1963
update tTableGramEpoch_tmp.tab set Col_1=1958-05-11 where Col_1=4-mar-1963
    has been executed
    update result of 1 rows
1 selected columns:  Col_1
 3.13926e+09
select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 2 rows
1 selected columns:  Col_1
 2.97562e+09
 3.13926e+09
using style python select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 2 rows
1 selected columns:  Col_1
04-Mar-1953
11-May-1958
select 34447 as TIME TIME giving tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  TIME
 34447
using style python select * from tTableGramEpoch_tmp.tab
    has been executed
    select result of 1 rows
1 selected columns:  TIME
11-Mar-1953

Structure of table /Users/diepen/casa/masktql/build/dbg/tables/TaQL/test/tTableGramEpoch_tmp.tab
------------------ 
1 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=StandardStMan  bucketsize=256
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=126
  TIME double   scalar unit=[d] measure=epoch,UTC

Keywords of main table 
----------------------
  Column TIME
    MEASINFO: {
      type: String "epoch"
      Ref: String "UTC"
    }
    QuantumUnits: String array with shape [1]
      [d]

create table tTableGramEpoch_tmp.tab TIME TIME
    has been executed
    cretab result of 0 rows
1 selected columns:  TIME

Structure of table /Users/diepen/casa/masktql/build/dbg/tables/TaQL/test/tTableGramEpoch_tmp.tab
------------------ 
0 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=StandardStMan  bucketsize=256
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  TIME double   scalar unit=[d] measure=epoch,UTC

Keywords of main table 
----------------------
  Column TIME
    MEASINFO: {
      type: String "epoch"
      Ref: String "UTC"
    }
    QuantumUnits: String array with shape [1]
      [d]

create table tTableGramEpoch_tmp.tab TIME EPOCH [UNIT="s"]
    has been executed
    cretab result of 0 rows
1 selected columns:  TIME

Structure of table /Users/diepen/casa/masktql/build/dbg/tables/TaQL/test/tTableGramEpoch_tmp.tab
------------------ 
0 rows, 1 columns in little endian format (using 1 data managers)

 StandardStMan file=table.f0  name=StandardStMan  bucketsize=256
    ActualCacheSize=2 PERSCACHESIZE=2 IndexLength=118
  TIME double   scalar unit=[s] measure=epoch,UTC

Keywords of main table 
----------------------
  Column TIME
    MEASINFO: {
      type: String "epoch"
      Ref: String "UTC"
    }
    QuantumUnits: String array with shape [1]
      [s]

